godtools/App/Share/Data/FavoritedResourcesRepository/Cache/RealmFavoritedResourcesCache.swift

```diff
@@ -90,38 +90,141 @@ class RealmFavoritedResourcesCache {
+                primaryKeyPath: #keyPath(RealmFavoritedResource.resourceId),
+                primaryKeys: [id]
+            )
+        }
+        .eraseToAnyPublisher()
+    }
+    
+    func reorderFavoritedResourcePublisher(id: String, originalPosition: Int, newPosition: Int) -> AnyPublisher<[FavoritedResourceDataModel], Error> {
+        
+        return realmDatabase.writeObjectsPublisher { realm in
+            var resourcesToUpdate: [RealmFavoritedResource] = []
+            
+            if newPosition - originalPosition > 0 {
+                let resourcesToMoveUp = realm.objects(RealmFavoritedResource.self).where({ $0.position >= originalPosition && $0.position <= newPosition })
+                for resource in resourcesToMoveUp {
+                    resource.position -= 1
+                }
+                
+                resourcesToUpdate += resourcesToMoveUp
+            } else {
+                let resourcesToMoveDown = realm.objects(RealmFavoritedResource.self).where( { $0.position <= originalPosition && $0.position >= newPosition})
+                for resource in resourcesToMoveDown {
+                    resource.position += 1
+                }
+                
+                resourcesToUpdate += resourcesToMoveDown
+            }
+            
+            if let resourceToMove = realm.object(ofType: RealmFavoritedResource.self, forPrimaryKey: id) {
+                resourceToMove.position = newPosition
+                resourcesToUpdate.append(resourceToMove)
+            }
+            
+            return resourcesToUpdate
+            
+        } mapInBackgroundClosure: { (objects: [RealmFavoritedResource]) in
             return objects.map({
                 FavoritedResourceDataModel(realmFavoritedResource: $0)
             })
         }
         .eraseToAnyPublisher()
     }
     
-    func deleteFavoritedResourcePublisher(id: String) -> AnyPublisher<Void, Error> {
+    // MARK: - Private
+    
+    @available(*, deprecated)
+    private func migrateExistingFavoritesToPositions(ascendingOrder: Bool) -> [RealmFavoritedResource] {
         
-        return realmDatabase.deleteObjectsInBackgroundPublisher(
-            type: RealmFavoritedResource.self,
-            primaryKeyPath: #keyPath(RealmFavoritedResource.resourceId),
-            primaryKeys: [id]
-        )
-        .eraseToAnyPublisher()
+        let realm = realmDatabase.openRealm()
+        let favoritesSortedByPosition = Array(
+            realm.objects(RealmFavoritedResource.self)
+            .sorted(byKeyPath: #keyPath(RealmFavoritedResource.position), ascending: ascendingOrder)
+            )
+        
+        if favoritesSortedByPosition.allSatisfy({ $0.position == 0 }) {
+            
+            let favoritesSortedByCreatedAt = realm
+                .objects(RealmFavoritedResource.self)
+                .sorted(byKeyPath: #keyPath(RealmFavoritedResource.createdAt), ascending: !ascendingOrder)
```

Comment:
Hey @rachaelblue I ran a fresh build on develop, pre migration, and added a tool to my favorited list.  I then ran this branch and I noticed my tool order was in reverse. 

It took me a second to see why, but right here on ascendingOrder the flag is getting flipped ```! ascendingOrder ```.

Reply:
I would suggest putting ```false``` there and removing the argument on the method.  We will always want to migrate those so the newest is first.

Reply:
Then you will need to update your methods ```getFavoritedResourcesSortedByPosition and getFavoritedResourcesSortedByPositionPublisher``` to sort by position if you want to use the ascendingOrder argument on those methods.

Reply:
Hey @rachaelblue that sounds good!